### PR TITLE
Autopilot - Identify fixes + Refactor

### PR DESCRIPTION
**Pull request by dhawal1248**

### CODE REVIEW

File: v0/destinations/autopilot/transform.js [hunk 1]
```diff
@@ -43,42 +46,35 @@ function responseBuilder(payload, message, autoPilotConfig) {
 }
 
 function getIdentifyPayload(message) {
-  const rawPayload = {};
   const contact = {};
+  let val;
 
   const traits = get(message.context.traits) ? message.context.traits : null;
```
Comment (arnab-p) [inline v0/destinations/autopilot/transform.js:52]:
```suggestion
  const traits = get(message.context.traits) ? message.context.traits : undefined;
```

File: v0/destinations/autopilot/transform.js [hunk 2]
```diff
@@ -43,42 +46,35 @@ function responseBuilder(payload, message, autoPilotConfig) {
...
 
   if (traits != null) {
```
Comment (arnab-p) [inline v0/destinations/autopilot/transform.js:54]:
```suggestion
  if (traits) {
```

File: v0/destinations/autopilot/transform.js [hunk 3]
```diff
@@ -27,6 +27,9 @@ function responseBuilder(payload, message, autoPilotConfig) {
       break;
     case EventType.TRACK:
       requestConfig = defaultPostRequestConfig;
+      if (!message.context.traits.email) {
```
Comment (ruchiramoitra) [inline v0/destinations/autopilot/transform.js:30]: I think we should give a check of whether message.context.traits exist other wise "error": "Cannot read property 'email' of undefined" will be thrown

**Review by ruchiramoitra — COMMENTED** [review]
updatePayload in transform.js is not used.
https://autopilot.docs.apiary.io/reference/api-methods/addupdate-contact.  ---> custom properties not being set this is the document.

**Comment by arnab-p** [comment]
Changes are done with another PR

**Pull request closed without merging**